lib/auth/auth.go

```diff
@@ -4200,13 +4207,48 @@ func (a *Server) UpsertDatabaseServer(ctx context.Context, server types.Database
 	return lease, nil
 }
 
+func (a *Server) UpdateOSSDesktopAlert(ctx context.Context) error {
+	windowsDesktops, err := a.GetWindowsDesktops(ctx, types.WindowsDesktopFilter{NonAD: []bool{true}})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	desktops := len(windowsDesktops)
```

Comment:
The idea is that we will let you add as many as you want, and they will all show up in the listing, but if you add > 3 you'll get an alert and connections to the desktops will fail.

This is because the registration/heartbeats typically happen by a machine, where an error message might go unnoticed by a user. If we push the error to connection time we can make a much nicer UX.

(It also solves the issue of what to do if you have 2 desktops and try to register 2 more - should we fail both of them, accept one but not the other, etc)